src/http/h1/encode.rs

```diff
@@ -107,50 +108,54 @@ impl Chunked {
             Chunked::End => return Ok(0),
             _ => {}
         }
+
         let mut n = {
-            let pieces = match *self {
+            // TODO: Find a better way to concatente
+            //    Could use write! if we reserved the required capacity upfront
+            match *self {
```

Comment:
The idea here is actually to try to use the `writev` syscall, which could be done with `AsyncWrite::write_bufs`. Before tokio, that's what `AtomicWrite` tried to do. `AtomicWrite` is now just a legacy artifact I haven't refactored yet.

The point is, it's better to keep all these as individual slices, and allow the `writev` to write all of them at once, without needing to copy them in user space.